### PR TITLE
Entitlements (V2) + support for Cognito app client with secret

### PR DESCRIPTION
**Pull request by eternelpanic**
So I needed an easy way to manipulate some entitlements, and your code was the easiest to start with, so Ive added in some support for both the AWS Cognito Secret Hash, and the Entitlements Service.
Its reasonable simple, but did the job for me allowing me to view and add permissions to users as required.

Others may find it useful so I have upstreamed it.

### CODE REVIEW

**Comment by puremcc** [comment]
@eternelpanic Thank you for this PR. I had been meaning to add Entitlements support for a while now, so this is much appreciated! If you don't mind just making two small fixes first, I will be glad to merge this in.

1. [critical] I appreciate the addition of the secret_hash for the case of App Client with Secret; however, we will need it to be a fully optional parameter, so that it doesn't break the alternative case of using the Default App Client (no secret). 
2. [low] For Python style purposes, `secretHash` should be `secret_hash`.

Below is a proposed resolution to both, as well as the integration test statement that should pass for both types of app clients.

Proposed update to `get_tokens` method in `osdu/client/aws.py`

```python
    def get_tokens(self, password, secret_hash) -> None:
        if self._profile:
            session = boto3.Session(profile_name=self._profile)
            print('Created boto3 session with profile: ', self._profile)
            cognito = session.client('cognito-idp')
        else:
            cognito = boto3.client('cognito-idp')

        auth_params = { 
            'USERNAME': self._user, 
            'PASSWORD': password
        }
        if secret_hash:
            auth_params['SECRET_HASH'] = secret_hash
        
        response = cognito.initiate_auth(
            AuthFlow='USER_PASSWORD_AUTH',
            ClientId=self._client_id,
            AuthParameters=auth_params
        )

        self._access_token = response['AuthenticationResult']['AccessToken']
        self._refresh_token = response['AuthenticationResult']['RefreshToken']
```

Integration test that should still pass without client secret:
_Note: I need to update the pipeline test instance so these tests aren't broken, but this should work against your own dev/test instance._

```sh
 python -m unittest -v tests.integration.TestAwsOsduClient
```

File: osdu/client/aws.py [hunk 1]
```diff
@@ -58,7 +60,7 @@ def get_tokens(self, password) -> None:
         response = cognito.initiate_auth(
```
Comment (puremcc) [inline osdu/client/aws.py:60]: In order to make `secret_hash` optional:

```suggestion
        auth_params = { 
            'USERNAME': self._user, 
            'PASSWORD': password
        }
        if secret_hash:
            auth_params['SECRET_HASH'] = secret_hash
        
        response = cognito.initiate_auth(
            AuthFlow='USER_PASSWORD_AUTH',
            ClientId=self._client_id,
            AuthParameters=auth_params
        )
```

File: osdu/client/aws.py [hunk 2]
```diff
@@ -41,13 +42,14 @@ def __init__(self, data_partition_id, api_url: str = None, client_id: str = None
...
-            self.get_tokens(password)
-            password = None  # Don't leave password lying around.
+            self.get_tokens(password, secret_hash)
+            password = None # Don't leave password lying around.
         else:
-            self.get_tokens(os.environ.get('OSDU_PASSWORD'))
+            self.get_tokens(os.environ.get('OSDU_PASSWORD'), secret_hash)
 
-    def get_tokens(self, password) -> None:
+    def get_tokens(self, password, secretHash) -> None:
```
Comment (puremcc) [inline osdu/client/aws.py:52]:
```suggestion
    def get_tokens(self, password, secret_hash) -> None:
```

**Review by puremcc — CHANGES_REQUESTED** [review]
Just need to make `secret_hash` optional. Currently passing null will raise a paramValidationError from botocore.

**Review by eternelpanic — COMMENTED** [review]
Add some return examples to the readme and correct the function names.

**Comment by eternelpanic** [comment]
Interestingly enough, an internal code review picked up the same issues. Your suggested changes are a little cleaner than my own, so I have merged them in.
I have also made some corrections to the readme and added some return value examples for the entitlement functions.

**Review by puremcc — APPROVED** [review]

**Comment by puremcc** [comment]
@eternelpanic I've approved the PR and updated its title to reflect its purpose. Please feel free to merge it in when you get a chance and consider summarizing the commit message appropriately.

On a side note, I have just pushed a small update in master to remove some of the fragile integration tests, so those CI checks will turn green going forward. You can pull from master for those changes first if you'd like or just merge your branch as-is; it's up to you. PR is approved either way 👍🏻

**Comment by puremcc** [comment]
@eternelpanic Apologies for the continued pipeline issues. Please go ahead and merge the PR. I'll get the pipeline fixed separately.

**Comment by eternelpanic** [comment]
Sorry, Im rather new to github (still love my svn), but my understanding is that I do not have access to merge the pull request.

`This branch has no conflicts with the base branch

Only those with write access to this repository can merge pull requests.`

**Comment by puremcc** [comment]
Ah, OK, I didn't realize that. I've merged the PR. Thanks again for your contribution! I will push the updated pip package shortly.

**Pull request merged**